Peter drake patch git

PeterDrake opened the pull request:
I copied the Git page over from Drakepedia.

alainkaegi commented:
Nice work.  A few comments/proposals:
- Is there a particular reason to have the log command show the full 40-character commit ID rather than the abbreviated version?  Personally, I use `git log --oneline --graph --all` that shows only as many digits as necessary to make each object distinct.
- We should add `gitlab.com` to make us not appear too biased.
- I propose that we annotate dangerous habits in bold (pairs of double-stars).  In particular, let's change Bad Idea to **bad idea**.

alainkaegi commented:
* git -> Git
* `git commit -am` -> `git commit -m 'Initial project version'`
* `git commit -am 'My poem is now twice as long'` -> `git add some-file; git commit -m 'My poem is now twice as long'` (put these commands on two separate lines)
* The output of `git log --oneline --graph --all` should use the abbreviated commit ids.
* If you get a vanilla merge conflict and type `git status`, Git will tell you to type  just `git commit` to complete the merge (not `git commit -a`). Perhaps there is value in our recommendations to follow Git's hints.
* Remote repositories: I prefer to run `git rebase` to incorporate recent changes in main:
```
git checkout main
git pull
git checkout experiment
git rebase main
```
Are you concerned that `git rebase` is too much for our students?  I personally think it's important to know. I think most serious shops would frown upon using `get merge` to incorporate recent changes into a branch.

PeterDrake commented:
I added the capitalization, fixed the log output, and removed the -a from the command to complete a conflicted merge.

I didn't separate adding from committing, as explained in the other pull request.

Yes, I avoid rebase. These students are going to make LOTS of Git mistakes at this stage. Creating "lost" commits (no longer on any branch) is a recipe for more confusion and frustration. I prefer keeping them around, even at the expense of a "messier" history. More advanced students (like those we bring on for summer research) can adopt such fancier features.

Maybe someday I'll add this later in the Software Development semester, after everyone has the basics down.

alainkaegi commented:
- Squashed related commits
- Edited commit messages
- Merge w/o fast-forwarding
